### PR TITLE
imgmath: Fix relative file path

### LINKED ISSUE #10944: imgmath failed to resolve image path for files in nested folder
### Describe the bug

Hi. I am using the sphinx.ext.imgmath plugin to generate images for inline math. The images (of math) are correctly generated in `build/html/_images`, but the generated HTML files do not contain correct link to them.

When there are nested files, instead of generating the link (to the image) relative to the file location, the generated links are instead relative to `build/html`. For example, say I have a page `source/foo/index.rst` that contains math. Instead of generating an image tag as `<img src="../_images/math/..."`, Sphinx generate ``<img src="_images/math/..."`` As a result, images of math are not displayed correctly.

This problem seems to be introduced in Sphinx [v5.3.0](https://github.com/sphinx-doc/sphinx/releases/tag/v5.3.0). [v5.2.3](https://github.com/sphinx-doc/sphinx/releases/tag/v5.2.3) does not have this behavior.

### How to Reproduce

Create the following 3 files:

index.rst
```rst
.. toctree::
   :maxdepth: 2
   :caption: Contents:
   :numbered:

   foo/index
```

foo/index.rst
```rst
.. math::
  1 + 1
```

conf.py
```
project = 'Test'
copyright = '2022, Lesley Lai'
author = 'Lesley Lai'

extensions = ['sphinx.ext.imgmath']

templates_path = ['_templates']
exclude_patterns = []

html_theme = 'alabaster'
html_static_path = ['_static']
```

Then run the following commands:
```bash
$ sphinx-build -M html . _build
$ # open _build/html/foo and see
```

### Environment Information

```text
Platform:              win32; (Windows-10-10.0.22623-SP0)
Python version:        3.10.0 (tags/v3.10.0:b494f59, Oct  4 2021, 19:00:18) [MSC v.1929 64 bit (AMD64)])
Python implementation: CPython
Sphinx version:        5.3.0
Docutils version:      0.16
Jinja2 version:        3.1.2
```

### Sphinx extensions

sphinx.ext.imgmath

### Additional context

To make `sphinx.ext.imgmath` work correctly, latex needs to be installed in the system.

### PR DESCRIPTION
**Pull request by jschueller**
Closes #10944

### CODE REVIEW

**Comment by jschueller** [comment]
@return42 and now ?

**Comment by return42** [comment]
Perfect .. works like a charm:

grafik

-----

IDK why the CI reports an [flake8 issue](https://github.com/sphinx-doc/sphinx/actions/runs/3600688863/jobs/6065620995#step:5:16) not related to this patch.

**Comment by jschueller** [comment]
lets hope it can be merged soon

**Comment by jschueller** [comment]
maybe @marxin or @jfbu ?

**Comment by jfbu** [comment]
@jschueller I rebased your work on current master tip so that we have fresh linting.

@AA-Turner Should we consider fixing the #10944 as done here for 6.0.1 rather than 6.1.0?

**Review by jfbu — COMMENTED** [review]
In reviewing this I realized #10888 modified the returned path from ``render_math()`` to be absolute rather than relative.  But the docstring still says https://github.com/sphinx-doc/sphinx/blob/a9b0f2708b67a355c5f4969ccbbe9bd695518805/sphinx/ext/imgmath.py#L212
Could the #10888 have been achieved while keeping the behaviour of ``render_math()`` as formerly? And at same time avoid #10944? If not the docstring of ``render_math()`` should be updated.

**Review by jfbu — CHANGES_REQUESTED** [review]
Actually #10888 also modified the returned type of ``render_math()`` but did not update accordingly the docstring, which still says that it also returns the temporary filepath as well as the absolute path of destination image file. I see no mention of this change in CHANGES or in our document. I think it is important that it should be documented somewhere (I don't know to what extent or if at all ``render_math()`` is public API. Edit: surely not public API, but my point is that docstrings should be kept up-to-date with code...).

**Comment by jschueller** [comment]
ok, I updated the docstring

File: sphinx/ext/imgmath.py [hunk 1]
```diff
@@ -209,10 +209,9 @@ def render_math(
     """Render the LaTeX math expression *math* using latex and dvipng or
     dvisvgm.
 
-    Return the filename relative to the built document and the "depth",
+    Return the absolute filename to the built document and the "depth",
```
Comment (jfbu) [inline sphinx/ext/imgmath.py]: I think you meant to remove "to the built document" part as it referred not to the image file but to the html one.

Comment (jschueller) [reply sphinx/ext/imgmath.py]: yes

**Review by jfbu — APPROVED** [review]
LGTM, thanks
ping @AA-Turner about perhaps adding some brief summary of overall architecture at start of module as the #10888 discussion was somewhat terse and it is hard to follow it a posteriori.  In  particular at 4cb857d5a65c07a49e1b8ece116d9880fae38152 the extension internal dynamics were somewhat refactored and it would be nice to have some brief overview in plain terms of what this all does.
